exercises/lens-person/src/Person.hs

```diff
@@ -1,49 +1,43 @@
-module Person where
-
-import           Data.Time.Calendar
-
-data Person = Person {
-      _name    :: Name,
-      _born    :: Born,
-      _address :: Address
-    }
-
-data Name = Name {
-      _foreNames :: String, -- Space separated
-      _surName   :: String
-    }
-
-data Born = Born {
-      _bornAt :: Address,
-      _bornOn :: Day
-    }
-
-data Address = Address {
-      _street      :: String,
-      _houseNumber :: Int,
-      _place       :: String, -- Village / city
-      _country     :: String
-    }
-
--- Valid values of Gregorian are those for which 'Data.Time.Calendar.fromGregorianValid'
--- returns Just.
-data Gregorian = Gregorian {
-      _year  :: Integer,
-      _month :: Int,
-      _day   :: Int
-    }
-
--- Implement these.
+module Person
+  ( Address (..)
+  , Born    (..)
+  , Name    (..)
+  , Person  (..)
+  , bornStreet
+  , renameStreets
+  , setBirthMonth
+  , setCurrentStreet
+  ) where
+
+import Data.Time.Calendar (Day)
+
+data Person = Person { _name    :: Name
+                     , _born    :: Born
+                     , _address :: Address
+                     }
+
+data Name = Name { _foreNames :: String
+                 , _surName   :: String
+                 }
+
+data Born = Born { _bornAt :: Address
+                 , _bornOn :: Day
+                 }
+
+data Address = Address { _street      :: String
+                       , _houseNumber :: Int
+                       , _place       :: String
+                       , _country     :: String
+                       }
 
 bornStreet :: Born -> String
 bornStreet = undefined
 
-setCurrentStreet :: String -> Person -> Person
-setCurrentStreet = undefined
+renameStreets :: (String -> String) -> Person -> Person
```

Comment:
Oh, I see. You made the stub file follow alphabetical order.

The old argument for having them in this order is that a plain set is probably easier than a transform (sorry that I forget the lens specific terminology), and if students implement from top to bottom, they should probably implement the easier ones first.

However, at this point the tests should be guiding the implementation order since the tests will be run in order and the first one will fail.

I'm okay with this, let me know if I missed anything in my reasoning and/or there are other concerns